biogrid/libs/biogrid-simulator/src/lib/community/ruralarea.ts

```diff
@@ -0,0 +1,44 @@
+import { Town } from '@biogrid/grid-simulator';
+import { Building } from '../building';
+
+/**
+ * A rural area that represents a community where a microgrid can function.
+ */
+export class RuralArea implements Town {
+
+  private town: Building[];
+
+  /**
+   * @param {Building[]} town A list of buildings which make up a town.
+   */
+  constructor(town: Building[]) {
+    this.town = town;
+  }
+
+  getTown() : Building[] {
+    return this.town;
+  }
+
+  /**
+   * This method gets a building in a town by its name identifier.
+   * @param {Building} building The building we're looking for.
+   * @return {Building} Returns the building if found in the list, null if not.
+   */
+  getBuildingByName(building: Building) {
+    for (let i=0; i<this.town.length; i++){
+      if (this.town[i].getBuildingName() === building.getBuildingName()){
+        return this.town[i];
+      }
+    }
+    return null;
+  }
+
+  /**
+   * This method adds a building to our town.
+   * @param {Building} newBuilding The building to be added. 
+   */
+  addBuilding(newBuilding: Building) {
+    this.town.push(newBuilding);
```

Comment:
If we're planning to use the building name as a key, it would be helpful to have some sort of checker in addBuilding to make sure the names stay unique. I think a better alternative would be to give buildings unique IDs instead of names. You could have the building object that is passed in as a parameter not require an id (have building.id unset), and when you call this function, it generates an ID, makes sure it doesn't already exist in the town (if ID already exists, keep generating until you find a unique one), assign the id to the building, and add the building (with id set) to the town. Then, this function returns the new Building object with the building.id set

Reply:
yea that makes sense. I'll implement it in an ID format.